docs/proposals/control-data-plane-split/README.md

```diff
@@ -0,0 +1,260 @@
+
+### Deployment Architecture
+
+![Deployment architecture](deployment-architecture.png)
+
+- _Control Plane Deployment_: The control plane is a Kubernetes Deployment with one container running the NGF
+  controller. The control plane will perform the same functions as it does today, but instead of
+  configuring nginx by writing files to a shared volume, it will send the configuration to the agent via gRPC.
+- _Control Plane Service_: Exposes the control plane via a Kubernetes Service of type `ClusterIP`. The data plane will
+  use the DNS name of the Service to connect to the control plane.
+- _Data Plane DaemonSet/Deployment_: The data plane can be deployed as either a DaemonSet or Deployment. The data
+  plane contains a single container running both the agent and nginx processes. The agent will download the
+  configuration from the control plane over a streaming RPC.
+- _NGINX Service_: Exposes nginx via a Kubernetes Service of type `LoadBalancer .` This is the entry point for the
+  customer’s traffic. Note that this Service should not expose any of the agent’s ports.
+
+#### Further Requirements and Implementation Details
+
+- Both deployments should have read only filesystems.
+- Both deployments should have the minimal permissions required to perform their functions.
+- The nginx deployment should be configurable via the helm chart.
+  - Downside of this is that these options will apply to all nginx instances.
+  - We could introduce a CRD, but where would it attach? We already have NginxProxy which controls dynamic data plane configuration, and this may eventually attach to the Gateway instead of just the GatewayClass. Would a Deployment configuration fit in there, and would it be dynamic? That would require us to completely redeploy nginx if a user changes those settings.
+  - We could start with the helm chart option, and rely on user feedback to see if we need to get more granular.
+  - This could also involve creating a ConfigMap that the control plane consumes on startup and contains all nginx Deployment/Daemonset configuration, including NGINX Plus usage configuration.
+- nginx Service should have configurable labels and annotations via the GatewayInfrastructure field in the Gateway resource.
+- Control plane creates the nginx deployment and service when a Gateway resource is created, in the same namespace as the Gateway resource. When the Gateway is deleted, the control plane deletes nginx deployment and service.
+- Control plane should label the nginx service and deployment with something related to the name of the Gateway so it can easily be linked.
+- Liveness/Readiness probes:
+  - Control plane probe currently waits until we configure nginx. Going forward, this probe should just be when the control plane is ready to configure.
+  - nginx probe...can agent expose any type of health endpoint? This is not yet clear.
+- Control plane should not restart data plane pods if they are unhealthy.
```

Comment:
Why this decision? Because the assumption is that the pod can only be unhealthy is the JWT is invalid? 

There can be other reasons for the pod to be unhealthy too due to memory issues and node eviction. Is this beyond the scope of this design?